packages/neon/neon/lib/src/platform/platform.dart

```diff
@@ -5,22 +5,24 @@ import 'package:meta/meta.dart';
 import 'package:neon/src/platform/android.dart';
 import 'package:neon/src/platform/linux.dart';
 
-Future<NeonPlatform> getNeonPlatform() async {
-  final NeonPlatform platform;
-  if (Platform.isAndroid) {
-    platform = const AndroidNeonPlatform();
-  } else if (Platform.isLinux) {
-    platform = const LinuxNeonPlatform();
-  } else {
-    throw UnimplementedError('No implementation for platform ${Platform.operatingSystem} found');
+@immutable
+abstract interface class NeonPlatform {
+  static late final NeonPlatform _platform;
+
+  static Future setup() async {
+    if (Platform.isAndroid) {
+      _platform = const AndroidNeonPlatform();
+    } else if (Platform.isLinux) {
+      _platform = const LinuxNeonPlatform();
+    } else {
+      throw UnimplementedError('No implementation for platform ${Platform.operatingSystem} found');
+    }
+
+    await _platform.init();
   }
 
-  await platform.init.call();
-  return platform;
-}
+  static NeonPlatform get instance => _platform;
```

Comment:
`_platform` should be nullable instead of late and we should throw if it's null (not yet set up).

And doc comments for class, setup function and getter please.

Reply:
Unit testing this will probably be hard :thinking:

Reply:
Can't we just use the same mocking approach as we used for TimerBloc?

Reply:
@Leptopoda ping

Reply:
Shure this makes sense for mocking.

I still think that we should throw if it hasn't been set up with a meaningful message like `NeonPlatform hasn't been set up yet. Please make sure NeonPlatform.setup() has been called before.`

Now if someone want's to access `NeonPlatform.instance` without having called setup before they just get a late init error which is harder to debug.

Reply:
Yes that's good!